File: 05.phase-2-output/backend/src/main/java/com/carddemo/entity/Account.java
package com.carddemo.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.math.BigDecimal;
import java.util.List;

@Entity
@Table(name = "accounts")
public class Account {

    @Id
    @NotNull
    @Column(name = "acct_id")
    private Long acctId;

    @NotNull
    @Size(max = 1)
    @Column(name = "acct_active_status", length = 1)
    private String acctActiveStatus;

    @Column(name = "acct_curr_bal", precision = 12, scale = 2)
    private BigDecimal acctCurrBal;

    @Column(name = "acct_credit_limit", precision = 12, scale = 2)
    private BigDecimal acctCreditLimit;

    @Column(name = "acct_cash_credit_limit", precision = 12, scale = 2)
    private BigDecimal acctCashCreditLimit;

    @Size(max = 10)
    @Column(name = "acct_open_date", length = 10)
    private String acctOpenDate;

    @Size(max = 10)
    @Column(name = "acct_expiraion_date", length = 10)
    private String acctExpiraionDate;

    @Size(max = 10)
    @Column(name = "acct_reissue_date", length = 10)
    private String acctReissueDate;

    @Column(name = "acct_curr_cyc_credit", precision = 12, scale = 2)
    private BigDecimal acctCurrCycCredit;

    @Column(name = "acct_curr_cyc_debit", precision = 12, scale = 2)
    private BigDecimal acctCurrCycDebit;

    @Size(max = 10)
    @Column(name = "acct_addr_zip", length = 10)
    private String acctAddrZip;

    @Size(max = 10)
    @Column(name = "acct_group_id", length = 10)
    private String acctGroupId;

    @OneToMany(mappedBy = "account", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Card> cards;

    @OneToMany(mappedBy = "account", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<TransactionCategoryBalance> transactionCategoryBalances;

    public Account() {}

    public Long getAcctId() {
        return acctId;
    }

    public void setAcctId(Long acctId) {
        this.acctId = acctId;
    }

    public String getAcctActiveStatus() {
        return acctActiveStatus;
    }

    public void setAcctActiveStatus(String acctActiveStatus) {
        this.acctActiveStatus = acctActiveStatus;
    }

    public BigDecimal getAcctCurrBal() {
        return acctCurrBal;
    }

    public void setAcctCurrBal(BigDecimal acctCurrBal) {
        this.acctCurrBal = acctCurrBal;
    }

    public BigDecimal getAcctCreditLimit() {
        return acctCreditLimit;
    }

    public void setAcctCreditLimit(BigDecimal acctCreditLimit) {
        this.acctCreditLimit = acctCreditLimit;
    }

    public BigDecimal getAcctCashCreditLimit() {
        return acctCashCreditLimit;
    }

    public void setAcctCashCreditLimit(BigDecimal acctCashCreditLimit) {
        this.acctCashCreditLimit = acctCashCreditLimit;
    }

    public String getAcctOpenDate() {
        return acctOpenDate;
    }

    public void setAcctOpenDate(String acctOpenDate) {
        this.acctOpenDate = acctOpenDate;
    }

    public String getAcctExpiraionDate() {
        return acctExpiraionDate;
    }

    public void setAcctExpiraionDate(String acctExpiraionDate) {
        this.acctExpiraionDate = acctExpiraionDate;
    }

    public String getAcctReissueDate() {
        return acctReissueDate;
    }

    public void setAcctReissueDate(String acctReissueDate) {
        this.acctReissueDate = acctReissueDate;
    }

    public BigDecimal getAcctCurrCycCredit() {
        return acctCurrCycCredit;
    }

    public void setAcctCurrCycCredit(BigDecimal acctCurrCycCredit) {
        this.acctCurrCycCredit = acctCurrCycCredit;
    }

    public BigDecimal getAcctCurrCycDebit() {
        return acctCurrCycDebit;
    }

    public void setAcctCurrCycDebit(BigDecimal acctCurrCycDebit) {
        this.acctCurrCycDebit = acctCurrCycDebit;
    }

    public String getAcctAddrZip() {
        return acctAddrZip;
    }

    public void setAcctAddrZip(String acctAddrZip) {
        this.acctAddrZip = acctAddrZip;
    }

    public String getAcctGroupId() {
        return acctGroupId;
    }

    public void setAcctGroupId(String acctGroupId) {
        this.acctGroupId = acctGroupId;
    }

    public List<Card> getCards() {
        return cards;
    }

    public void setCards(List<Card> cards) {
        this.cards = cards;
    }

    public List<TransactionCategoryBalance> getTransactionCategoryBalances() {
        return transactionCategoryBalances;
    }

    public void setTransactionCategoryBalances(List<TransactionCategoryBalance> transactionCategoryBalances) {
        this.transactionCategoryBalances = transactionCategoryBalances;
    }

}

File: 05.phase-2-output/backend-nodejs/src/models/Account.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Account = sequelize.define('Account', {
  acctId: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    allowNull: false,
    field: 'acct_id'
  },
  acctActiveStatus: {
    type: DataTypes.STRING(1),
    allowNull: false,
    validate: {
      isIn: [['A', 'C']]
    },
    field: 'acct_active_status'
  },
  acctCurrBal: {
    type: DataTypes.DECIMAL(12, 2),
    allowNull: false,
    defaultValue: 0,
    field: 'acct_curr_bal'
  },
  acctCreditLimit: {
    type: DataTypes.DECIMAL(12, 2),
    allowNull: false,
    field: 'acct_credit_limit'
  },
  acctCashCreditLimit: {
    type: DataTypes.DECIMAL(12, 2),
    field: 'acct_cash_credit_limit'
  },
  acctOpenDate: {
    type: DataTypes.STRING(10),
    field: 'acct_open_date'
  },
  acctExpiraionDate: {
    type: DataTypes.STRING(10),
    field: 'acct_expiraion_date'
  },
  acctReissueDate: {
    type: DataTypes.STRING(10),
    field: 'acct_reissue_date'
  },
  acctCurrCycCredit: {
    type: DataTypes.DECIMAL(12, 2),
    field: 'acct_curr_cyc_credit'
  },
  acctCurrCycDebit: {
    type: DataTypes.DECIMAL(12, 2),
    field: 'acct_curr_cyc_debit'
  },
  acctAddrZip: {
    type: DataTypes.STRING(10),
    field: 'acct_addr_zip'
  },
  acctGroupId: {
    type: DataTypes.STRING(10),
    field: 'acct_group_id'
  }
}, {
  tableName: 'accounts',
  timestamps: true
});

module.exports = Account;
